### PR TITLE
WIP Fix bug where record key wouldn’t be updated with actual ID

### PR DESCRIPTION
**Pull request by mattvague**
Previously the create success reducer would update the record, but not update the record key if using a cid for the map store. Would end up with something like this:

```
{
  ‘myCid123’: {
	id: 1234,
	...
  }
}
```

Also cleaned up the create success reducer quite a bit. All tests pass and everything seems to work in my app, but I'm still a little worried maybe I missed something when replacing the previous logic?

BTW @sporto Thank you for adding the Map store!!!

### CODE REVIEW

File: src/reducers/map/create/success.ts [hunk 1]
```diff
@@ -14,31 +14,15 @@ var invariantArgs: InvariantsBaseArgs = {
...
 	invariants(invariantArgs, config, current, addedRecord)
 
-	var key = config.key
-	var done = false
-	var addedRecordKey = addedRecord[key]
-
-	// Update existing records
-	var updatedCollection = r.map((existingRecord) => {
-		var recordKey = existingRecord[key]
-		if (recordKey == null) throw new Error('Expected record to have ' + key)
```
Comment (mattvague) [inline src/reducers/map/create/success.ts]: @sporto Seemed weird to me to check here that previously inserted records have keys, maybe this should be added to the create start reducer instead?

Comment (mattvague) [inline src/reducers/map/create/success.ts]: Come to think of it.... I'm wondering why the "it throws when record doesnt have an id" test didn't start failing when I removed this. Hmmmm

File: src/reducers/map/create/success.ts [hunk 2]
```diff
@@ -14,31 +14,15 @@ var invariantArgs: InvariantsBaseArgs = {
...
+  var addedRecordKey = addedRecord[key]
+
+  var addedRecordKeyLens = r.lensProp(addedRecordKey)
+  var clientGenKeyLens = r.lensProp(clientGenKey)
+
+  if (r.view(clientGenKeyLens, current)) {
+    return r.set(addedRecordKeyLens, addedRecord, r.dissoc(clientGenKey, current))
+  }
+
+  return r.set(addedRecordKeyLens, addedRecord, current)
```
Comment (mattvague) [inline src/reducers/map/create/success.ts]: Feel free to rip apart my ramda skillz here if there's some way to make this simpler. Very much a beginner!

**Comment by sporto** [comment]
Hi, thanks for doing this. I will merge this and publish a patch as the current reducer is broken.
I'll look at the details with more time later.

**Comment by sporto** [comment]
On second thought, will not publish straight away.  I will take some time later today or tomorrow to look at what needs to happen here. Feel free to put another PR to improve this.

**Comment by sporto** [comment]
@mattvague 	I have done some changes. Please try version 3.0.2 thanks

**Pull request merged**